GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -153,11 +155,21 @@ internal static Process StartProcess(string fileName, string arguments, string w
             startProcess.EnableRaisingEvents = true;
             startProcess.Exited += (sender, args) =>
             {
+                var executionEndTimestamp = DateTime.Now;
                 string quotedCmd = fileName;
                 if (quotedCmd.IndexOf(' ') != -1)
                     quotedCmd = quotedCmd.Quote();
-                var executionEndTimestamp = DateTime.Now;
-                AppSettings.GitLog.Log(quotedCmd + " " + arguments, executionStartTimestamp, executionEndTimestamp);
+
+                //GitHub #4213 Commands are duplicated in GE Gitcommand log
+                string duplicate = "";
+                if (_lastLogArguments == arguments && (_lastLogTimestamp - executionStartTimestamp).TotalMilliseconds == 0)
+                {
+                    duplicate = " #Duplicate?";
+                }
+                _lastLogArguments = arguments;
+                _lastLogTimestamp = executionStartTimestamp;
+
+                AppSettings.GitLog.Log(quotedCmd + " " + arguments + duplicate, executionStartTimestamp, executionEndTimestamp);
```

Comment:
can't we simply unwire the event once it fired?

```
        internal static Process StartProcess(string fileName, string arguments, string workingDirectory, Encoding outputEncoding)
        {
            // ...
            startProcess.EnableRaisingEvents = true;
            EventHandler processExited = null;
            processExited = (sender, args) =>
            {
                startProcess.Exited -= processExited;

                string quotedCmd = fileName;
                if (quotedCmd.IndexOf(' ') != -1)
                    quotedCmd = quotedCmd.Quote();
                var executionEndTimestamp = DateTime.Now;
                AppSettings.GitLog.Log(quotedCmd + " " + arguments, executionStartTimestamp, executionEndTimestamp);
            };
            startProcess.Exited += processExited;

            return startProcess;
        }
```

Here's the result:
![image](https://user-images.githubusercontent.com/4403806/33831731-5b4093b8-decd-11e7-9b4c-5cdf0edb9e5b.png)

Reply:
> can't we simply unwire the event once it fired?

Seem like a better solution. I will need to read on the details before I "approve". 
@RussKie  as you did the solution, I can "approve" a PR from you instead.